server/src/main/java/org/elasticsearch/action/search/TransportSearchAction.java

```diff
@@ -1011,6 +1016,7 @@ static List<SearchShardIterator> getRemoteShardsIteratorFromPointInTime(
                 remoteShardIterators.add(shardIterator);
             }
         }
+        assert contextsMap.isEmpty() : "shards that were in the PIT id were not returned by search shards?";
```

Comment:
Search shards runs now the can match phase, hence depending on the specific query, some of the shards may be skipped which are part of the PIT id. Those though would still be returned as part of the search shards response and have the flag `skip` set to `true`. That should mean that this assertion is valid.